src/main/scala/com/nec/arrow/ArrowInterfaces.scala

```diff
@@ -311,4 +320,20 @@ object ArrowInterfaces {
     intVector.clear()
     intVector.close()
   }
+
+  def nullable_bigint_vector_to_TimeStampVector(
+    input: nullable_bigint_vector,
+    timeStampVector: TimeStampMicroTZVector
+  ): Unit = {
+    if ( input.count < 1 ) {
+      return
+    }
+    timeStampVector.setValueCount(input.count)
+    getUnsafe.copyMemory(
+      input.validityBuffer,
+      timeStampVector.getValidityBufferAddress,
+      Math.ceil(input.count / 64.0).toInt * 8
+    )
+    getUnsafe.copyMemory(input.data, timeStampVector.getDataBufferAddress, input.size())
```

Comment:
which `.size()` is this?

Reply:
The same as with bigint_vector.  Should we also rename this to dataSize()?

Reply:
Yes would be a good idea to do that

Reply:
Changed size() to dataSize() in the others.